storm-core/src/jvm/backtype/storm/topology/WindowedBoltExecutor.java

```diff
@@ -0,0 +1,200 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static backtype.storm.topology.base.BaseWindowedBolt.Count;
+import static backtype.storm.topology.base.BaseWindowedBolt.Duration;
+
+/**
+ * An {@link IWindowedBolt} wrapper that does the windowing of tuples.
+ */
+public class WindowedBoltExecutor implements IRichBolt {
+    private static final Logger LOG = LoggerFactory.getLogger(WindowedBoltExecutor.class);
+
+    private IWindowedBolt bolt;
+    private transient WindowedOutputCollector windowedOutputCollector;
+    private transient WindowLifecycleListener<Tuple> listener;
+    private transient WindowManager<Tuple> windowManager;
+
+    public WindowedBoltExecutor(IWindowedBolt bolt) {
+        this.bolt = bolt;
+    }
+
+    private int getTopologyTimeoutMillis(Map stormConf) {
+        if (stormConf.containsKey(Config.TOPOLOGY_ENABLE_MESSAGE_TIMEOUTS)) {
+            boolean timeOutsEnabled = (boolean) stormConf.get(Config.TOPOLOGY_ENABLE_MESSAGE_TIMEOUTS);
+            if (!timeOutsEnabled) {
+                return Integer.MAX_VALUE;
+            }
+        }
+        int timeout = 0;
+        if (stormConf.containsKey(Config.TOPOLOGY_MESSAGE_TIMEOUT_SECS)) {
+            timeout = ((Number) stormConf.get(Config.TOPOLOGY_MESSAGE_TIMEOUT_SECS)).intValue();
+        }
+        return timeout * 1000;
+    }
+
+    private void ensureDurationLessThanTimeout(int duration, int timeout) {
+        if (duration > timeout) {
+            throw new IllegalArgumentException("Window duration (length + sliding interval) value " + duration +
+                                                       " is more than " + Config.TOPOLOGY_MESSAGE_TIMEOUT_SECS +
+                                                       " value " + timeout);
+        }
+    }
+
+    // TODO: add more validation
+    private void validate(Map stormConf, Count windowLengthCount, Duration windowLengthDuration,
+                          Count slidingIntervalCount, Duration slidingIntervalDuration) {
+
+        int topologyTimeout = getTopologyTimeoutMillis(stormConf);
+        if (windowLengthDuration != null && slidingIntervalDuration != null) {
+            ensureDurationLessThanTimeout(windowLengthDuration.value + slidingIntervalDuration.value, topologyTimeout);
+        } else if (windowLengthDuration != null) {
+            ensureDurationLessThanTimeout(windowLengthDuration.value, topologyTimeout);
+        }
+    }
+
+    private WindowManager<Tuple> initWindowManager(WindowLifecycleListener<Tuple> lifecycleListener, Map stormConf) {
+        WindowManager<Tuple> manager = new WindowManager<>(lifecycleListener);
+        Duration windowLengthDuration = null;
+        Count windowLengthCount = null;
+        Duration slidingIntervalDuration = null;
+        Count slidingIntervalCount = null;
+        if (stormConf.containsKey(Config.TOPOLOGY_BOLTS_WINDOW_LENGTH_COUNT)) {
```

Comment:
Given we are getting these from storm config, does this mean a user can not have 2 bolts with different windowing intervals?